### PR TITLE
fix: StreamingResponse compatibility issue

### PR DESCRIPTION
**Pull request by xianml**
## What does this PR address?
fix issue:
Exception raised when calling bentoml.client with a Multipart format. 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/paperspace/.local/lib/python3.9/site-packages/bentoml/_internal/client/__init__.py", line 133, in _sync_call
    return asyncio.run(self._call(inp, _bentoml_api=_bentoml_api, **kwargs))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/paperspace/.local/lib/python3.9/site-packages/bentoml/_internal/client/http.py", line 167, in _call
    fake_resp = await api.input.to_http_response(kwargs, None)
  File "/home/paperspace/.local/lib/python3.9/site-packages/bentoml/_internal/io_descriptors/multipart.py", line 294, in to_http_response
    return await concat_to_multipart_response(dict(zip(self._inputs, resps)), ctx)
  File "/home/paperspace/.local/lib/python3.9/site-packages/bentoml/_internal/utils/formparser.py", line 371, in concat_to_multipart_response
    writer.write(resp.body)
AttributeError: 'StreamingResponse' object has no attribute 'body'
```

The reason is that we directly change the output of Text descriptor to `StreamingResponse` from `Response`, while it may not compatible with `concat_to_multipart_response` func of  multi-part descriptor

this fix is to return `StreamingResponse` or `Response`  according to the input.

test:
case 1: in linux 
image

case 2: in colab
image

Fixes #(issue)

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [x] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [x] Did you write tests to cover your changes?

### CODE REVIEW

**Comment by xianml** [comment]
@aarnphm @bojiang  hi, do you have any time for this PR?

**Comment by aarnphm** [comment]
@xianml seems to me like we need to fix the multipart parser instead of patching the text descriptor right?

**Comment by xianml** [comment]
> @xianml seems to me like we need to fix the multipart parser instead of patching the text descriptor right?

Yeah. fixed in `concat_to_multipart_response` .

File: tests/unit/_internal/io/test_multipart.py [hunk 1]
```diff
@@ -11,8 +11,10 @@
 from bentoml.io import JSON
 from bentoml.io import Image
 from bentoml.io import Multipart
+from bentoml.io import Text
 
 example = Multipart(arg1=JSON(), arg2=Image(mime_type="image/bmp", pilmode="RGB"))
+example2 = Multipart(arg1=Image(), arg2=Text())
```
Comment (aarnphm) [inline tests/unit/_internal/io/test_multipart.py:17]:
```suggestion
example2 = Multipart(arg1=Image(), arg2=Text(media_type='text/event-stream'))
```

**Pull request merged**